interfaces/policy/basedeclaration_test.go

```diff
@@ -830,16 +830,50 @@ plugs:
 }
 
 func (s *baseDeclSuite) TestOpticalDriveWrite(c *C) {
-	const plugYaml = `name: plug-snap
+	type options struct {
+		readonlyYamls []string
+		writableYamls []string
+	}
+
+	opts := &options{
+		readonlyYamls: []string{
+			// Undefined "write" attribute
+			`name: plug-snap
+version: 0
+plugs:
+  optical-drive: {}
+`,
+			// False "write" attribute
+			`name: plug-snap
+version: 0
+plugs:
+  optical-drive:
+    write: false
+`,
+		},
```

Comment:
Thanks for these tests!

While it is covered elsewhere, for clarity of this test, can you add:
```
			// Non-specified "write" attribute
			`name: plug-snap
version: 0
plugs:
  optical-drive: null
`,